hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestPipelineManagerImpl.java

```diff
@@ -0,0 +1,466 @@
+    dbStore = DBStoreBuilder.createDBStore(conf, new SCMDBDefinition());
+    nodeManager = new MockNodeManager(true, 20);
+    eventQueue = new EventQueue();
+    maxPipelineCount = nodeManager.getNodeCount(HddsProtos.NodeState.HEALTHY) *
+        conf.getInt(OZONE_DATANODE_PIPELINE_LIMIT,
+            OZONE_DATANODE_PIPELINE_LIMIT_DEFAULT) /
+        HddsProtos.ReplicationFactor.THREE.getNumber();
+  }
+
+  @After
+  public void cleanup() throws Exception {
+    if (dbStore != null) {
+      dbStore.close();
+    }
+    FileUtil.fullyDelete(testDir);
+  }
+
+  private PipelineManagerV2Impl createPipelineManager()
+      throws IOException {
+    return PipelineManagerV2Impl.newPipelineManager(
+        conf, MockSCMHAManager.getInstance(),
+        nodeManager,
+        SCMDBDefinition.PIPELINES.getTable(dbStore), eventQueue);
+  }
+
+  @Test
+  public void testCreatePipeline() throws Exception {
+    List<Pipeline> pipelines = new ArrayList<>();
+    PipelineManagerV2Impl pipelineManager = createPipelineManager();
+    Assert.assertTrue(pipelineManager.getPipelines().isEmpty());
+    pipelineManager.allowPipelineCreation();
+    Pipeline pipeline1 = pipelineManager.createPipeline(
+        HddsProtos.ReplicationType.RATIS, HddsProtos.ReplicationFactor.THREE);
+    Assert.assertEquals(1, pipelineManager.getPipelines().size());
+    Assert.assertTrue(pipelineManager.containsPipeline(pipeline1.getId()));
+    pipelines.add(pipeline1);
+
+    Pipeline pipeline2 = pipelineManager.createPipeline(
+        HddsProtos.ReplicationType.RATIS, HddsProtos.ReplicationFactor.ONE);
+    Assert.assertEquals(2, pipelineManager.getPipelines().size());
+    Assert.assertTrue(pipelineManager.containsPipeline(pipeline2.getId()));
+    pipelines.add(pipeline2);
+    pipelineManager.close();
+
+    PipelineManagerV2Impl pipelineManager2 = createPipelineManager();
+    // Should be able to load previous pipelines.
+    Assert.assertFalse(pipelineManager.getPipelines().isEmpty());
+    Assert.assertEquals(2, pipelineManager.getPipelines().size());
+    pipelineManager.allowPipelineCreation();
+    Pipeline pipeline3 = pipelineManager.createPipeline(
+        HddsProtos.ReplicationType.RATIS, HddsProtos.ReplicationFactor.THREE);
+    Assert.assertEquals(3, pipelineManager.getPipelines().size());
+    Assert.assertTrue(pipelineManager.containsPipeline(pipeline3.getId()));
+    pipelines.add(pipeline3);
+
+    pipelineManager2.close();
+  }
+
+  @Test
+  public void testUpdatePipelineStates() throws Exception {
+    PipelineManagerV2Impl pipelineManager = createPipelineManager();
+    pipelineManager.allowPipelineCreation();
+    Pipeline pipeline = pipelineManager.createPipeline(
+        HddsProtos.ReplicationType.RATIS, HddsProtos.ReplicationFactor.THREE);
+    Assert.assertEquals(1, pipelineManager.getPipelines().size());
+    Assert.assertTrue(pipelineManager.containsPipeline(pipeline.getId()));
+    Assert.assertTrue(pipeline.getPipelineState() == ALLOCATED);
```

Comment:
`Assert.assertTrue` can be replaced with `Assert.assertEquals`.
`Assert.assertEquals` will give a better context in case of test failure.

Reply:
These are copied from original TestSCMPipelineManager tests. I will update them.